src/main/java/com/karankumar/bookproject/backend/entity/RatingScale.java

```diff
@@ -62,13 +66,13 @@ public String toString() {
         return rating;
     }
 
-    public static Double toDouble(RatingScale ratingScale) {
+    public static Optional<Double> toDouble(RatingScale ratingScale) {
         if (ratingScale == null) {
-            LOGGER.log(Level.INFO, "Returning null as rating was null");
-            return null;
+            LOGGER.log(Level.INFO, "Returning optional as rating was null");
+            return empty();
         }
 
-        return ratingScale.value;
+        return ofNullable(ratingScale.value);
```

Comment:
I guess this method can still return double and if scale is null we can return 0.0. I can see we are calling this method at lots of places and assigning a default value of 0.0

Reply:
> if scale is null we can return 0.0

On our [rating scale](https://github.com/Project-Books/book-project/blob/master/src/main/java/com/karankumar/bookproject/backend/entity/RatingScale.java#L29), we have 0.0, but we also have an enum value for `NO_RATING`, which should be the default